### ISSUE TITLE
Queue is not a constructor

### ISSUE THREAD

**Issue by sfreytag** [mourner/rbush-knn#18]
When using rbush-knn with webpack, I get `Queue is not a constructor` from line 14 of index.js:

```
var queue = new Queue(undefined, compareDist);
```

 At this point, Queue is a module. I can fix it for my use by changing the top of the file to:

```
'use strict';

import Queue  from 'tinyqueue';

export default function knn(tree, x, y, n, predicate, maxDistance) {
```

then Queue is a class and `new Queue` works fine

I'm fairly new to using modules. I don't know if this is the correct fix , but if it looks useful I can send a PR for it.

**Comment by sfreytag** [comment]
I've pushed the code that's using it up to https://github.com/sfreytag/projfinder so here's a bit more context.

My tweaked version of knn-rbush that works with my app lives in the file:

https://github.com/sfreytag/projfinder/blob/master/src/map/knn.js

It's imported in MapApp.vue:

https://github.com/sfreytag/projfinder/blob/master/src/components/map/MapApp.vue

If I just import knn from the one installed via npm then I get the constructor error mentioned above.  So you can replicate this if you clone my repo, do an `npm install rbush-knn` then tweak MapApp.vue to `import knn from "knn"` instead of from `../../map/knn.js`

**Comment by sfreytag** [comment]
I cut my code right down to a simple example to help me work through this, which I've put here:

https://github.com/sfreytag/rbushknn-testing

This is a webpack project that creates an RBush tree and searches it using knn-rbush. If I open the built webpage in a browser I get the `Queue is not a constructor` error.

This discussion proved quite helpful, which I found via a closed issue on tinyqueue about IE11 and ES6 modules:
https://github.com/mourner/flatbush/issues/23

That helped me work out I can fix the problem with this webpack config:
```
  resolve: {
    alias: {
      "tinyqueue": __dirname + "/node_modules/tinyqueue/tinyqueue.js"
    }
  }
```

Or a similar alias to point rbush-knn at `/node_modules/rbush_knn/rbush_knn.js`. And this fix also works in the larger Vue app. So maybe that will do the job. 

Two other fixes that worked:

1. change rbush-knn's `package.json` `main` key from `index.js` to `rbush-knn.js`
2. change rbush-knn index.js:3 to `var Queue = require('tinyqueue').default;`
 
In summary it would seem this is indeed Webpack glitch, preferring to load from `module` rather than `main` when tinyqueue is required rather than imported. Quite probably this bug: https://github.com/webpack/webpack/issues/5756

So fixing it in my Vue or Webpack config is probably fine for now.
 
However would you be interested in a pull request to convert rbush-knn to an ES6 module, with builds for ES5 for the `main` along the lines of the discussion in https://github.com/mourner/flatbush/issues/23 ? As per the hack I posted at the top I think that would also fix my problem because webpack would be OK with the import of tinyqueue rather than require. It would be cleaner for me than other fixes mentioned here. And it perhaps might be a useful tweak for this project?

**Comment by smaant** [comment]
I stumbled upon the same problem yesterday, thank you very much @sfreytag for the workaround with webpack config, worked like a charm!

**Comment by chimurai** [comment]
Thanks @sfreytag. Your solution pointed me to a solution for Angular:

`tsconfig.json`

```json
{
  "compilerOptions": {
      "tinyqueue": ["node_modules/tinyqueue/tinyqueue.js"]
    }
}
```

**Comment by JackCA** [comment]
thanks for the work @chimurai and @mourner -- any chance we could get a release on this?

**Comment by catdad** [comment]
This issue was fixed in 2020 yet still has not been released in 2023. Is there anything the community can do to help with this?
